scaling/scaling.go

```diff
@@ -0,0 +1,111 @@
+package scaling
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/e2etests/scaling/provider"
+)
+
+type Config struct {
+	Logger   micrologger.Logger
+	Provider provider.Interface
+}
+
+type Scaling struct {
+	logger   micrologger.Logger
+	provider provider.Interface
+}
+
+func New(config Config) (*Scaling, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Provider == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Provider must not be empty", config)
+	}
+
+	s := &Scaling{
+		logger:   config.Logger,
+		provider: config.Provider,
+	}
+
+	return s, nil
+}
+
+func (s *Scaling) Test(ctx context.Context) error {
```

Comment:
The steps being processed here are also just taken from the `aws-operator` test.